packages/core/src/browser/shell/description-widget.ts

```diff
@@ -0,0 +1,29 @@
+/********************************************************************************
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+ import { Emitter } from '../../common/event';
+
+export interface DescriptionWidget {
+    description: string;
+    onDescriptionChange: Emitter<void>;
+}
+
+export namespace DescriptionWidget {
+    /* eslint-disable @typescript-eslint/no-explicit-any */
+    export function is(arg: any): arg is DescriptionWidget {
+        return arg !== undefined && !!arg['onDescriptionChange'];
+    }
```

Comment:
Suggestion: We generally try not to use `any`, even in `is` methods. Change this to `Object | undefined` or `unknown`.

Reply:
I copied this implementation from a few examples in the rest of the code and did initially try and use `unknown`, but the compiler wouldn't let me use the string object accessor. Will update to `Object | undefined`.

Reply:
For `unknown` you'll have to do this:

```ts
return typeof arg === 'object' && arg && !!arg['onDescriptionChange'];
```

Reply:
I still get compiler errors about any having an implicit any type whichever way I try this..

Reply:
@aaronrhodes you can do the following to avoid using `any`:

```ts
export namespace DescriptionWidget {
    export function is(arg: Object | undefined): arg is DescriptionWidget {
        return !!arg && typeof arg === 'object' && 'onDidDescriptionChange' in arg;
    }
}
```

Reply:
👍